### PR TITLE
Cxtb 409 automate employee health portal review your upcoming appointments and calendar for employee ehp user accounts

### PR DESCRIPTION
**Pull request by jmontero-nts**
Automated test case:
[https://digitalscientists.atlassian.net/browse/CXTB-409](https://digitalscientists.atlassian.net/browse/CXTB-409)

Added test.
Added Page Objects
Added helper test
Proof Of Successful run:
image

### CODE REVIEW

File: tests/cases/employee_health_portal/case_employee_health_portal_home.yaml
Comment (csanchez-nts) [inline tests/cases/employee_health_portal/case_employee_health_portal_home.yaml:1]: You need to move this test to the correct file

**Review by csanchez-nts — CHANGES_REQUESTED** [review]

File: tests/cases/employee_health_portal/case_employee_health_portal_home.yaml
Comment (jmontero-nts) [reply tests/cases/employee_health_portal/case_employee_health_portal_home.yaml:1]: I already move it.

File: tests/cases/employee_health_portal/case_employee_health_portal_request_appointment.yaml [hunk 1]
```diff
@@ -75,50 +67,170 @@ TestEmployeeHealthPortalRequestAnAppointmentForUser:
...
+    # look for another day without even scheduled a date.
+    - Type: generate_random_day
+      ResultVar: Day
+    # checking a RandomDay
     - Type: click
       Strategy: xpath
-      Id: $PAGE.employee_health_telehealth_request_form.calendar_right_button$
+      Id: $PAGE.employee_health_home_page.calendar_day_button$
+    - Type: sleep
+      Time: 1
```
Comment (miguel-hquez) [inline tests/cases/employee_health_portal/case_employee_health_portal_request_appointment.yaml:167]: is this sleep needed for a reason?

File: tests/cases/employee_health_portal/case_employee_health_portal_request_appointment.yaml [hunk 2]
```diff
@@ -75,50 +67,170 @@ TestEmployeeHealthPortalRequestAnAppointmentForUser:
...
+      Id: $PAGE.employee_health_home_page.container_tab$
+  #validating upcoming scheduled event
+    - Type: wait_for
+      Strategy: xpath
+      Id: $PAGE.employee_health_home_page.upcoming_appointment_title$
+    - Type: get_attribute
+      Strategy: xpath
+      Id: $PAGE.employee_health_home_page.upcoming_appointment_title$
+      Greps:
+        - var: MANYAPPOINTMENTS
```
Comment (miguel-hquez) [inline tests/cases/employee_health_portal/case_employee_health_portal_request_appointment.yaml:127]: please stick with the format separating words using '_'

File: tests/cases/employee_health_portal/case_employee_health_portal_request_appointment.yaml [hunk 3]
```diff
@@ -75,50 +67,170 @@ TestEmployeeHealthPortalRequestAnAppointmentForUser:
...
+    - Type: generate_random_day
+      ResultVar: Day
+    # checking a RandomDay
     - Type: click
       Strategy: xpath
-      Id: $PAGE.employee_health_telehealth_request_form.create_event_button$
-  # Verify if the appointment is showing on the list
+      Id: $PAGE.employee_health_home_page.calendar_day_button$
+    - Type: sleep
+      Time: 1
```
Comment (miguel-hquez) [inline tests/cases/employee_health_portal/case_employee_health_portal_request_appointment.yaml:199]: is this sleep needed?

File: tests/cases/employee_health_portal/case_employee_health_portal_request_appointment.yaml [hunk 4]
```diff
@@ -75,50 +67,170 @@ TestEmployeeHealthPortalRequestAnAppointmentForUser:
...
+      Id: $PAGE.employee_health_home_page.calendar_day_button$
+    # look for another day without even scheduled a date.
+    - Type: generate_random_day
+      ResultVar: Day
+    # checking a RandomDay
+    - Type: click
+      Strategy: xpath
+      Id: $PAGE.employee_health_home_page.calendar_day_button$
+    - Type: sleep
+      Time: 1
```
Comment (miguel-hquez) [inline tests/cases/employee_health_portal/case_employee_health_portal_request_appointment.yaml:230]: is this sleep needed

File: tests/page_objects/employee_health_portal_page.yaml [hunk 5]
```diff
@@ -43,6 +43,20 @@ employee_health_home_page:
...
+  tabs_no_results: //div[@class = 'tabs-no-results']
+  view_upcoming_appointment_button: //button[text()='View upcoming appointments']
+  schedule_tab_button: //button[@class = 'box-tabs-nav-item active']
+  container_tab: //div[contains(@class, 'box has-scrollable-content is-white has-shadow')]
+  first_day: (//time[contains(text(), $AND_CLI_NEXT_MONTH$)])[1]
+  second_day: (//time[contains(text(), $AND_CLI_NEXT_MONTH$)])[2]
+  third_day: (//time[contains(text(), $AND_CLI_NEXT_MONTH$)])[3]
+
+
+
```
Comment (miguel-hquez) [inline tests/page_objects/employee_health_portal_page.yaml:59]: could you delete this unused new lines in the middle of the file?

File: tests/cases/employee_health_portal/case_employee_health_portal_request_appointment.yaml (hunk 2 above)
Comment (jmontero-nts) [reply tests/cases/employee_health_portal/case_employee_health_portal_request_appointment.yaml]: Done

File: tests/cases/employee_health_portal/case_employee_health_portal_request_appointment.yaml (hunk 3 above)
Comment (jmontero-nts) [reply tests/cases/employee_health_portal/case_employee_health_portal_request_appointment.yaml]: yes, when I use wait_for  the test failure trying to click on View upcoming appointment button.

File: tests/cases/employee_health_portal/case_employee_health_portal_request_appointment.yaml (hunk 4 above)
Comment (jmontero-nts) [reply tests/cases/employee_health_portal/case_employee_health_portal_request_appointment.yaml:295]: yes, when I use wait_for  the test failure trying to click on View upcoming appointment button.

File: tests/page_objects/employee_health_portal_page.yaml (hunk 5 above)
Comment (jmontero-nts) [reply tests/page_objects/employee_health_portal_page.yaml]: Done

File: lib/core/device.rb [hunk 6]
```diff
@@ -1933,6 +1933,20 @@ def return_element_attribute(action)
 
 end
 
+# Recieve a timestamp and return the related day
+def get_day(action)
```
[the remark this answers is not in the document]
Comment (jmontero-nts) [reply lib/core/device.rb:1937]: it has been added.

File: lib/core/device.rb [hunk 7]
```diff
@@ -1933,6 +1933,20 @@ def return_element_attribute(action)
...
+  timestamp_string = convert_value(action["Timestamp"])
+  timestamp = timestamp_string.to_i
+  time = Time.at(timestamp/1000)
+  ENV[convert_value(action["ResultVar"])] = time.day.to_s
+end
+
+# Return a random day
+def generate_random_day(action)
```
[the remark this answers is not in the document]
Comment (jmontero-nts) [reply lib/core/device.rb:1953]: it has been added as well.

File: README.md [hunk 8]
```diff
@@ -1163,4 +1163,25 @@ Guarantee that the checkbox is checked or unchecked depending on the option.
     - Type: credentials_checkbox
       Strategy: xpath | id | etc.
       Option: check | uncheck
-      Id: //some//path
\ No newline at end of file
+      Id: //some//path
+
+### <a id="generate_random_day"></a>generate_random_day
+
+obtain a random day but 31 is not included
```
[the remark this answers is not in the document]
Comment (jmontero-nts) [reply README.md]: resolved in this commit https://github.com/neveralone-chs-org/TestRay/pull/122/commits/e762203a3887abdb970c7e8a704e2134b208634b

File: README.md [hunk 9]
```diff
@@ -1163,4 +1163,25 @@ Guarantee that the checkbox is checked or unchecked depending on the option.
...
+
+Recieve a timestamp and return the related day
+
+    - Type: get_day
+      Timestamp: $AND_CLI_FIRST_SCHEDULED_DAY$
+      ResultVar: DAY
+
+### <a id="get_next_month"></a>get_next_month
+
+return the following month of the current date
```
[the remark this answers is not in the document]
Comment (jmontero-nts) [reply README.md]: resolved in this commit https://github.com/neveralone-chs-org/TestRay/pull/122/commits/e762203a3887abdb970c7e8a704e2134b208634b

**Review by csanchez-nts — CHANGES_REQUESTED** [review]

**Review by csanchez-nts — DISMISSED** [review]

**Review by csanchez-nts — APPROVED** [review]

**Review by miguel-hquez — APPROVED** [review]

**Pull request merged**